File: CSharp/Solutions/9.1. Problems For Champions - Part I/MagicDates.cs
namespace MagicDates
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using System.Threading.Tasks;
    
    public class MagicDates
    {
        public static void Main(string[] args)
        {
            int startYear = int.Parse(Console.ReadLine());
            int endYear = int.Parse(Console.ReadLine());
            int magicWeight = int.Parse(Console.ReadLine());

            DateTime currentDate = new DateTime(startYear, 1, 1);

            bool printed = false;

            while (currentDate.Year <= endYear)
            {
                int dateWeight = 0;
                string dateString = currentDate.ToString("ddMMyyyy");

                for (int i = 0; i < dateString.Length - 1; i++)
                {
                    for (int j = i + 1; j < dateString.Length; j++)
                    {
                        dateWeight += (dateString[i] - '0') * (dateString[j] - '0');
                    }
                }

                if (dateWeight == magicWeight)
                {
                    Console.WriteLine(currentDate.ToString("dd-MM-yyyy"));
                    printed = true;
                }

                currentDate = currentDate.AddDays(1);
            }

            if (!printed)
            {
                Console.WriteLine("No");
            }
        }
    }
}

File: Java/Solutions/9.1. Problems For Champions - Part I/MagicDates.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Scanner;

public class MagicDates {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int startYear = Integer.parseInt(scanner.nextLine());
        int endYear = Integer.parseInt(scanner.nextLine());
        int magicWeight = Integer.parseInt(scanner.nextLine());

        LocalDate currentDate = LocalDate.of(startYear, 1, 1);

        boolean printed = false;
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("ddMMyyyy");
        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");

        while (currentDate.getYear() <= endYear) {
            int dateWeight = 0;
            String dateString = formatter.format(currentDate);

            for (int i = 0; i < dateString.length() - 1; i++) {
                for (int j = i + 1; j < dateString.length(); j++) {
                    dateWeight += (dateString.charAt(i) - '0') * (dateString.charAt(j) - '0');
                }
            }

            if (dateWeight == magicWeight) {
                System.out.println(outputFormatter.format(currentDate));
                printed = true;
            }

            currentDate = currentDate.plusDays(1);
        }

        if (!printed) {
            System.out.println("No");
        }
    }
}
